[hist] Clang-Tidy Clazy warnings

For example, this one looks like a real bug:
`/opt/root_src/hist/hist/src/AnalyticalIntegrals.cxx:54:11: warning: Called C++ object pointer is null [clang-analyzer-core.CallAndMessage]
`

Full report:
[hist.txt](https://github.com/root-project/root/files/6148218/hist.txt)

--- a/hist/hist/src/AnalyticalIntegrals.cxx
+++ b/hist/hist/src/AnalyticalIntegrals.cxx
@@ -9,7 +9,7 @@
  **********************************************************************/
 //
 //  AnalyticalIntegrals.cxx
-//  
+//
 //
 //  Created by Aurélie Flandi on 09.09.14.
 //
@@ -38,8 +38,8 @@ Double_t AnalyticalIntegral(TF1 *f, Double_t a, Double_t b)
 
    TFormula * formula = f->GetFormula();
    if (!formula) {
-      Error("TF1::AnalyticalIntegral","Invalid formula number - return a NaN"); 
-      result = TMath::QuietNaN();
+      Error("TF1::AnalyticalIntegral","Invalid formula number - return a NaN");
+      return TMath::QuietNaN();
    }
 
    if   (num == 200)//expo: exp(p0+p1*x)
@@ -64,9 +64,9 @@ Double_t AnalyticalIntegral(TF1 *f, Double_t a, Double_t b)
       double mean  = p[1];
       double sigma = p[2];
       //printf("computing integral for landau in [%f,%f] for m=%f s = %f \n",xmin,xmax,mean,sigma);
-      if (formula->TestBit(TFormula::kNormalized) ) 
+      if (formula->TestBit(TFormula::kNormalized) )
          result = amp*(ROOT::Math::landau_cdf(xmax,sigma,mean) - ROOT::Math::landau_cdf(xmin,sigma,mean));
-      else 
+      else
          result = amp*sigma*(ROOT::Math::landau_cdf(xmax,sigma,mean) - ROOT::Math::landau_cdf(xmin,sigma,mean));
    }
    else if (num == 500) //crystal ball
@@ -76,9 +76,9 @@ Double_t AnalyticalIntegral(TF1 *f, Double_t a, Double_t b)
       double sigma = p[2];
       double alpha = p[3];
       double n     = p[4];
-      
+
       //printf("computing integral for CB in [%f,%f] for m=%f s = %f alpha = %f n = %f\n",xmin,xmax,mean,sigma,alpha,n);
-      if (alpha > 0) 
+      if (alpha > 0)
          result = amp*( ROOT::Math::crystalball_integral(xmin,alpha,n,sigma,mean) -  ROOT::Math::crystalball_integral(xmax,alpha,n,sigma,mean) );
       else {
          result = amp*( ROOT::Math::crystalball_integral(xmax,alpha,n,sigma,mean) -  ROOT::Math::crystalball_integral(xmin,alpha,n,sigma,mean) );
@@ -95,6 +95,6 @@ Double_t AnalyticalIntegral(TF1 *f, Double_t a, Double_t b)
    }
    else
       result = TMath::QuietNaN();
-   
+
    return result;
 }

--- a/hist/hist/src/HFitImpl.cxx
+++ b/hist/hist/src/HFitImpl.cxx
@@ -658,7 +658,7 @@ void HFit::StoreAndDrawFitFunction(FitObject * h1, TF1 * f1, const ROOT::Fit::Da
          funcList->Add(fnew3);
       }
       else {
-         fnew2 = dynamic_cast<TF3*>(f1);
+         fnew3 = dynamic_cast<TF3*>(f1);
          R__ASSERT(fnew3);
       }
       fnew3->SetRange(xmin,ymin,zmin,xmax,ymax,zmax);

--- a/hist/hist/src/TF1.cxx
+++ b/hist/hist/src/TF1.cxx
@@ -3062,14 +3062,15 @@ TH1   *TF1::DoCreateHistogram(Double_t xmin, Double_t  xmax, Bool_t recreate)
       // delete previous histograms if were done if done in different mode
       xtitle = fHistogram->GetXaxis()->GetTitle();
       ytitle = fHistogram->GetYaxis()->GetTitle();
-      if (!gPad->GetLogx()  &&  fHistogram->TestBit(TH1::kLogX)) {
+      Bool_t test_logx = fHistogram->TestBit(TH1::kLogX);
+      if (!gPad->GetLogx() && test_logx) {
          delete fHistogram;
-         fHistogram = 0;
+         fHistogram = nullptr;
          recreate = kTRUE;
       }
-      if (gPad->GetLogx()  && !fHistogram->TestBit(TH1::kLogX)) {
+      if (gPad->GetLogx() && !test_logx) {
          delete fHistogram;
-         fHistogram = 0;
+         fHistogram = nullptr;
          recreate = kTRUE;
       }
    }

--- a/hist/hist/src/TFormula.cxx
+++ b/hist/hist/src/TFormula.cxx
@@ -1388,11 +1388,11 @@ void TFormula::HandleFunctionArguments(TString &formula)
          } else {
             // otherwise, try to match default parametrized functions
 
-            for (auto keyval : parFunctions) {
+            for (const auto &keyval : parFunctions) {
                // (name, ndim)
-               pair<TString, Int_t> name_ndim = keyval.first;
+               const pair<TString, Int_t> &name_ndim = keyval.first;
                // (formula without normalization, formula with normalization)
-               pair<TString, TString> formulaPair = keyval.second;
+               const pair<TString, TString> &formulaPair = keyval.second;
 
                // match names like gaus, gausn, breitwigner
                if (name == name_ndim.first)

--- a/hist/hist/src/TH1Merger.cxx
+++ b/hist/hist/src/TH1Merger.cxx
@@ -592,6 +592,7 @@ void TH1Merger::CopyBuffer(TH1 *hsrc, TH1 *hdes)
       //void *p2 = hdes ? hdes->fBuffer : 0;
       //Warning("TH1Merger::CopyMerge", "invalid inputs: %p, %p, %p, %p -> do nothing", hsrc, hdes, p1, p2);
       Warning("TH1Merger::CopyMerge", "invalid inputs: %p, %p, %p, -> do nothing", hsrc, hdes, p1);
+      return;
    }
 
    // Entries from buffers have to be filled one by one

--- a/hist/histpainter/src/THistPainter.cxx
+++ b/hist/histpainter/src/THistPainter.cxx
@@ -3402,7 +3402,7 @@ void THistPainter::ExecuteEvent(Int_t event, Int_t px, Int_t py)
    if (!gPad) return;
 
    static Int_t bin, px1, py1, px2, py2, pyold;
-   static TBox *zoombox;
+   static TBox *zoombox = nullptr;
    Double_t zbx1,zbx2,zby1,zby2;
 
    Int_t bin1, bin2;
@@ -3468,6 +3468,7 @@ void THistPainter::ExecuteEvent(Int_t event, Int_t px, Int_t py)
             zby1 = TMath::Power(10,zby1);
             zby2 = TMath::Power(10,zby2);
          }
+         if (zoombox) Error("ExecuteEvent", "Last zoom box was not deleted");
          zoombox = new TBox(zbx1, zby1, zbx2, zby2);
          Int_t ci = TColor::GetColor("#7d7dff");
          TColor *zoomcolor = gROOT->GetColor(ci);
@@ -3534,8 +3535,10 @@ void THistPainter::ExecuteEvent(Int_t event, Int_t px, Int_t py)
             zby2 = gPad->AbsPixeltoY(py);
             if (gPad->GetLogx()) zbx2 = TMath::Power(10,zbx2);
             if (gPad->GetLogy()) zby2 = TMath::Power(10,zby2);
-            zoombox->SetX2(zbx2);
-            zoombox->SetY2(zby2);
+            if (zoombox) {
+               zoombox->SetX2(zbx2);
+               zoombox->SetY2(zby2);
+            }
             gPad->Modified();
             gPad->Update();
          }
@@ -3607,7 +3610,7 @@ void THistPainter::ExecuteEvent(Int_t event, Int_t px, Int_t py)
                yaxis->SetRangeUser(y1, y2);
             }
             zoombox->Delete();
-            zoombox = 0;
+            zoombox = nullptr;
          }
       }
       gPad->Modified(kTRUE);

